odoo/modules/db.py

```diff
@@ -191,3 +212,482 @@ def has_trigram(cr: BaseCursor) -> bool:
+
+        subprocess.run(
+            [
+                find_pg_tool('psql'),
+                '--dbname', db_name,
+                '--file', os.path.join(dump_dir, 'dump.sql'),
+                '--quiet',
+            ],
+            env=exec_pg_environ(),
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.STDOUT,
+            check=True,
+        )
+
+        registry = odoo.modules.registry.Registry.new(db_name)
+        with registry.cursor() as cr:
+            env = odoo.api.Environment(cr, odoo.api.SUPERUSER_ID, {})
+            if copy:
+                # if it's a copy of a database, force generation of a new dbuuid
+                env['ir.config_parameter'].init(force=True)
+            if neutralize_database:
+                odoo.modules.neutralize.neutralize_database(cr)
+
+            if filestore_path:
+                filestore_dest = env['ir.attachment']._filestore()
+                shutil.move(filestore_path, filestore_dest)
+
+    _logger.info("RESTORE DB: %s", db_name)
+
+
+def rename(
+    admin_passwd: str | _SkipAdminPassword,
+    old_name: str,
+    new_name: str,
+) -> None:
+    verify_db_management_enabled()
+    verify_admin_password(admin_passwd)
+    odoo.modules.registry.Registry.delete(old_name)
+    odoo.sql_db.close_db(old_name)
+
+    db = odoo.sql_db.db_connect('postgres')
+    with closing(db.cursor()) as cr:
+        # database-altering operations cannot be executed inside a transaction
+        cr._cnx.autocommit = True
+        _drop_conn(cr, old_name)
+        try:
+            cr.execute(SQL(
+                'ALTER DATABASE %s RENAME TO %s',
+                make_database_identifier(cr, old_name),
+                make_database_identifier(cr, new_name),
+            ))
+        except psycopg2.Error as e:
+            e.add_note(f"Renaming {old_name} to {new_name}")
+            raise
+        _logger.info("RENAME DB: %s -> %s", old_name, new_name)
+
+    old_fs = odoo.tools.config.filestore(old_name)
+    new_fs = odoo.tools.config.filestore(new_name)
+    if os.path.exists(old_fs) and not os.path.exists(new_fs):
+        shutil.move(old_fs, new_fs)
+
+
+def list_dbs(*, force=False):
+    if not force:
+        verify_db_management_enabled()
+
+    if not odoo.tools.config['dbfilter'] and odoo.tools.config['db_name']:
+        # In case --db-filter is not provided and --database is passed, Odoo will not
+        # fetch the list of databases available on the postgres server and instead will
+        # use the value of --database as comma seperated list of exposed databases.
+        return sorted(odoo.tools.config['db_name'])
+
+    chosen_template = odoo.tools.config['db_template']
+    templates_list = tuple({'postgres', chosen_template})
+    db = odoo.sql_db.db_connect('postgres')
+    with closing(db.cursor()) as cr:
+        try:
+            cr.execute("""
+                SELECT datname
+                  FROM pg_database
+                 WHERE datdba = (
```

Comment:
The SQL indentation could be better ;)
(commit: [REF] core: database management functions)